src/charm.py

```diff
@@ -110,20 +123,19 @@ def __init__(self, *args) -> None:  # type: ignore[no-untyped-def]
             self._on_all_certificates_invalidated,
         )
 
+        self.framework.observe(self._ingress_provider.on.data_provided, self._on_data_provided)
+        self.framework.observe(self._ingress_provider.on.data_removed, self._on_data_removed)
+
     @property
     def _labels(self) -> dict[str, str]:
         """Get labels assigned to resources created by this app."""
         return {CREATED_BY_LABEL: self.app.name}
 
     @validate_config_and_integration(defer=False)
-    def _reconcile(self) -> None:
-        """Reconcile charm status based on configuration and integrations.
-
-        Raises:
-            RuntimeError: when initializing the lightkube client fails,
-            or when creating the gateway resource fails.
-        """
+    def _reconcile(self) -> None:  # pylint: disable=too-many-locals
+        """Reconcile charm status based on configuration and integrations."""
```

Comment:
I think the description should be more explicit and describe the steps executed in this method

Reply:
I refactored this method to be more straigh-forward to understand and also updated the docstring to include more information